src/create-reminder-signup/lib/identity.ts

```diff
@@ -0,0 +1,43 @@
+import type { Response as FetchResponse } from 'node-fetch';
+import fetch from 'node-fetch';
+import { isProd } from '../../lib/stage';
+
+const idapiBaseUrl = isProd()
+	? 'https://idapi.theguardian.com'
+	: 'https://idapi.code.dev-theguardian.com';
+
+const encodeEmail = (email: string): string =>
+	encodeURI(email).replace('+', '%2B');
+
+export function getIdentityIdByEmail(
+	email: string,
+	accessToken: string,
+): Promise<string> {
+	return fetch(`${idapiBaseUrl}/user?emailAddress=${encodeEmail(email)}`, {
+		headers: { 'X-GU-ID-Client-Access-Token': `Bearer ${accessToken}` },
+	})
+		.then((response: FetchResponse) => {
+			if (response.status == 404) {
+				return Promise.reject(new Error(`Email not found: ${email}`));
+			}
+			if (!response.ok) {
+				return Promise.reject(
+					new Error(
+						`Identity API user endpoint responded with status: ${response.status}`,
+					),
+				);
+			}
+			return response.json();
+		})
+		.then((identityResponse) => {
+			if (identityResponse.user?.id) {
+				return Promise.resolve(identityResponse?.user?.id as string);
+			} else {
+				return Promise.reject(
+					`Missing brazeUuid in identity API response: ${JSON.stringify(
+						identityResponse,
+					)}`,
+				);
```

Comment:
Should this one refect with an error? Maybe these need to be different types of erros, so that we can distinguish in the caller?

Reply:
I'm never sure what the best way of doing this is in javascript!
You can construct an `Error` object with a `name`, e.g.
`const error = (name: string, message: string): Error => ({message, name});`
So e.g. the `name` could indicate whether it 404d, or something else.

And then the caller can do something like:
```
getIdentityIdByEmail(...)
  .then(writeToDb)
  .catch(err => {
    if (err.name === '404') ...
    else ...
  })
```

An alternative would be to encode the possibility of failure in the return type, e.g. `Promise<IdentityId | IdentityError>`, but this could get quite verbose